### DISCUSSION TITLE
How does the search work?

### DISCUSSION THREAD

**Discussion by ovizii** [thlucas1/spotifyplus_card#14] [Q&A] [answered]
I clicked on search, but I don't see any field where I could enter text for the search.

![image](https://github.com/user-attachments/assets/9ddf9d73-57ce-4766-a5a1-f9b53d5a5b9c)

**Comment by thlucas1** [comment]
Are you running HA 2024.10 release?  If not, you will need to upgrade the HA release to fix this.

The SpotifyPlus search uses the HA `<ha-md-button-menu>` control to display the dropdown list.  The control was renamed from `<ha-button-menu-new>` to `<ha-md-button-menu>` with the HA 2024.10 release ([PR #22016](https://github.com/home-assistant/frontend/commit/d2194d55f9f6de15cdede6ebee95e2a959fe57fc)).  

The reason the menu is displayed like it is in your screenshot is because the card cannot find the `<ha-md-button-menu>` control, which does not exist in HA 2024.09.x (e.g. pre HA 2024.10) release.

This took me by surprise as well, as they just renamed the control out of the blue!

**Comment by thlucas1** [comment, marked as answer]
@ovizii 
Thanks for bringing this to my attention - it confused me as well when I upgraded to the 2024.10 release.  I just forgot to document it.

I created a new [TroubleShooting Guide](https://github.com/thlucas1/spotifyplus_card/wiki/Troubleshooting-Guide) page in the wiki for issues like this, as I am sure you are not the only one running on a pre 2024.10 release.  

Loving the feedback - keep it coming, and thank you for helping me out.

**Reply by ovizii** [reply]
I am on the latest HA version though.

![image](https://github.com/user-attachments/assets/7f006a2c-fabf-475f-b28c-c5c3e04614af)

**Reply by ovizii** [reply]
Apologies, you were right. Additionally, it was a browser caching issue. Maybe drop a hint in the wiki too, to definitely do a forced browser refresh.

**Reply by thlucas1** [reply]
Noted.  Updated troubleshooting doc with the following:

__How do I fix it?__  
- Upgrade to the latest HA 2024.10 (or greater) release.
- Ensure SpotifyPlus v1.0.6 or greater is installed.
- Be sure to do a "forced refresh" of the page, to ensure any cached pages are not displayed.

Thank you!

**Comment by thlucas1** [comment]
The Search Media browser should look like the following.  The "Albums, Artists, etc" is a dropdown list that will appear when you click the "Search for x" box.

![image](https://github.com/user-attachments/assets/6baf8aec-b155-42bd-b98b-7c36a3f4b8aa)
